src/collection-preferences/content-display/index.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { useUniqueId } from '../../internal/hooks/use-unique-id';
+
+import { CollectionPreferencesProps } from '../interfaces';
+import styles from '../styles.css.js';
+import { getSortedOptions, OptionWithVisibility } from './utils';
+import { DndContext } from '@dnd-kit/core';
+import { arrayMove, SortableContext, verticalListSortingStrategy } from '@dnd-kit/sortable';
+import { SortableItem } from './sortable-item';
+import useDragAndDropReorder from './use-drag-and-drop-reorder';
+import useLiveAnnouncements from './use-live-announcements';
+
+const componentPrefix = 'content-display';
+
+const getClassName = (suffix: string) => styles[`${componentPrefix}-${suffix}`];
+
+interface ContentDisplayPreferenceProps extends CollectionPreferencesProps.ContentDisplayPreference {
+  onChange: (value: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>) => void;
+  value?: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>;
+}
+
+export default function ContentDisplayPreference({
+  title,
+  description,
+  options,
+  value = options.map(({ id }) => ({
+    id,
+    visible: true,
+  })),
+  onChange,
+  liveAnnouncementDndStarted,
+  liveAnnouncementDndItemReordered,
+  liveAnnouncementDndItemCommitted,
+  liveAnnouncementDndDiscarded,
+  dragHandleAriaDescription,
+  dragHandleAriaLabel,
+}: ContentDisplayPreferenceProps) {
+  const idPrefix = useUniqueId(componentPrefix);
+
+  const onToggle = (option: OptionWithVisibility) => {
+    onChange(value.map(item => (item.id === option.id ? { ...item, visible: !option.visible } : item)));
+  };
+
+  const titleId = `${idPrefix}-title`;
+  const descriptionId = `${idPrefix}-description`;
+
+  const sortedOptions = getSortedOptions({ options, contentDisplay: value });
+
+  const { activeItem, collisionDetection, handleKeyDown, isKeyboard, sensors, setActiveItem } = useDragAndDropReorder({
+    sortedOptions,
+  });
+
+  const announcements = useLiveAnnouncements({
+    isDragging: activeItem !== null,
+    liveAnnouncementDndStarted,
+    liveAnnouncementDndItemReordered,
+    liveAnnouncementDndItemCommitted,
+    liveAnnouncementDndDiscarded,
+    sortedOptions: value,
+  });
+
+  return (
+    <div className={styles[componentPrefix]}>
+      <h3 className={getClassName('title')} id={titleId}>
+        {title}
+      </h3>
+      <p className={getClassName('description')} id={descriptionId}>
+        {description}
+      </p>
```

Comment:
As in the visible content preference, the header is an h3 but visually looks like the labels of other preferences such as page size, which are much smaller than any header you can make with the `Header` component. Also, `Header` adds extra padding. So this would force us to write overriding CSS, so that we would be worse off. Also, we would go against our linting rules that disallow using `className` and `id` in components.

Regarding the description, we could save _one_ line of CSS by using `Box`, at the expense of one line of JS to import it, and again the linting, so not really much benefit to get from here...